### DISCUSSION TITLE
Monorepo with local dependencies

### DISCUSSION THREAD

**Discussion by hanselabreu** [vercel/vercel#5167] [Help]
Hey guys,

I made a simple monorepo project using Lerna and I am having trouble deploying it.

I have two `packages` one called `web` and the other `components`. This is the `package.json` of the `web` package:
```
{
  "name": "web",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start"
  },
  "dependencies": {
    "next": "9.5.2",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "components": "file:../components"
  }
}
```

As you can see I am referencing the `components` locally but getting the following error on Vercel:

```
11:53:52.612 | Cloning completed in 814ms
11:53:52.613 | Analyzing source code...
11:53:54.706 | Installing build runtime...
11:53:55.173 | Build runtime installed: 466.836ms
11:53:56.137 | Installing dependencies...
11:53:57.289 | npm ERR! code ENOLOCAL
11:53:57.290 | npm ERR! Could not install from "../components" as it does not contain a package.json file.
11:53:57.294 | npm ERR! A complete log of this run can be found in:
11:53:57.294 | npm ERR!     /vercel/.npm/_logs/2020-09-08T15_53_57_291Z-debug.log
11:53:57.305 | Error: Command "npm install" exited with 1
11:53:59.145 | Done with "package.json"
```

The directory `components` does includes the following `package.json`: 

```
{
  "name": "components",
  "version": "1.0.0",
  "main": "dist/index.js",
  "source": "lib/index.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1",
    "dev": "microbundle watch --jsx React.createElement lib/*.js",
    "build": "microbundle --jsx React.createElement lib/*.js"
  },
  "devDependencies": {
    "microbundle": "^0.12.3"
  },
  "peerDependencies": {
    "react": "^16.13.1"
  }
}
```

But not sure as to what could be happening.

I guess the question is, am I even doing this right? The goal would be to not have to create `npm packages`.

Thanks guys :)

**Comment by styfle** [comment]
It sounds like you're describing Yarn Workspaces.

You should be able to change `"components": "file:../components"` to `"components": "1.2.3"` or whatever the version is in `../components/package.json`.

Then it will work anywhere `yarn install` works.

Remember to add the `workspaces` key to the root package.json like so: https://github.com/vercel/vercel/blob/290fdc0506f868851bf49ffd79d5643ce86283bc/package.json#L6-L13
